CRM/Contribute/BAO/Contribution.php

```diff
@@ -5669,4 +5674,79 @@ public static function replaceContributionTokens(
     return $contributionDetails;
   }
 
+  /**
+   * Record adjusted amount.
+   *
+   * @param int $updatedAmount
+   * @param int $paidAmount
+   * @param int $contributionId
+   *
+   * @param int $taxAmount
+   * @param bool $updateAmountLevel
+   *
+   * @return bool|\CRM_Core_BAO_FinancialTrxn
+   */
+  public static function recordAdjustedAmt($updatedAmount, $paidAmount, $contributionId, $taxAmount = NULL, $updateAmountLevel = NULL) {
```

Comment:
How is the change in the price of related line item(s) indicated? Without that, I don't see how one can create appropriate entity_financial_trxn transactions? Also, just want to check that $taxAmount is being calculated at line_item/financial_item layer and bubbled up to contribution layer. If so, can you change the documentation to indicate these presuppositions of this function?